Added docs for new RPM repos

fheinecke opened the pull request:
Update the YUM install commands to point to the new YUM repos as a part of RFD 0058.

Related PR: https://github.com/gravitational/teleport/pull/14203

Commit:
Added docs for new RPM repos

r0mant commented on docs/pages/includes/install-linux.mdx:
```diff
@@ -22,12 +22,14 @@
+        # file for each major release of Teleport.
+        # Note: if using a fork of RHEL/CentOS or Amazon Linux you may need to use '$ID_LIKE'
+        # and the codename your distro was forked from instead of '$ID'
+        $ sudo yum-config-manager --add $(rpm --eval "https://yum.releases.teleport.dev/$ID/$VERSION_ID/Teleport/%{_arch}/stable/v10/teleport.repo")
```
```suggestion
        $ sudo yum-config-manager --add $(rpm --eval "https://yum.releases.teleport.dev/$ID/$VERSION_ID/Teleport/%{_arch}/stable/v(=teleport.major_version=)/teleport.repo")
```

r0mant replied:
Also, what's the difference between `--add-repo` flag that was being used here before, and `--add` which is being used now?

fheinecke replied:
Effectively no difference. `yum-config-manager` expands `--add` to `--add-repo`. Updated the flag to the full value just in case there is some weird compat issue with one of the older OS versions we support.

Commit:
Fixed typos

webvictim commented on docs/pages/includes/install-linux.mdx:
````diff
@@ -87,6 +89,17 @@
         ```
   </TabItem>
 
+  <TabItem label="Amazon Linux 2/RHEL Legacy (RPM)">
````
Are we planning on maintaining this old repo? I almost think we should remove the instructions to avoid confusion (some people just won't read the "(Legacy)" in the header)

r0mant replied:
We'll probably be maintaining old repos (both apt and rpm) for foreseeable future. We thought about removing these instructions but don't want to scare people who still depend on them. Maybe adding a more prominent warning/admonition on this tab would be appropriate - saying that these are legacy ones and you should use the other repo but these aren't going anywhere. cc @fheinecke

r0mant commented:
@ptgott Can you take a look at this please?

ptgott reviewed (approved):
The "supported versions" link 404s for me. Also, it looks like supported OS versions are [spread out](https://github.com/gravitational/teleport/blob/2d66b8603304e115a2b0dee67a9f71bad346fccc/build.assets/tooling/cmd/build-os-package-repos/runners.go) across a few declarations, so linking to the source might be a bit inconvenient for users.

Maybe we can add a table above or below the relevant code snippets?

If we're pressed for time, please:
- Remove the "Supported versions are listed here" line from the first code snippet
- Let me know and I can create a docs issue to add context and list supported OS versions

fheinecke committed:
Added/fixed supported version links

fheinecke commented:
> The "supported versions" link 404s for me. Also, it looks like supported OS versions are [spread out](https://github.com/gravitational/teleport/blob/2d66b8603304e115a2b0dee67a9f71bad346fccc/build.assets/tooling/cmd/build-os-package-repos/runners.go) across a few declarations, so linking to the source might be a bit inconvenient for users.

Thanks for catching that. I've updated the APT link and added one for YUM. The declarations are spread out into two structs, but they are os package manager specific. Now that there are two separate links directly tied to what package manager/OS a customer is using I think they can be treated independently.

@ptgott Do you still want me to add a table? If so, do you want it in the docs or in the code? I had originally linked to the code as it is the source of truth. If we're going to add table based off that information it doesn't matter to me where it goes.

fheinecke committed:
Merge branch 'master' into fred/new-yum-docs

fheinecke committed:
Updated docs to use gpg key in new APT repos

ptgott commented:
@fheinecke I think this is okay to merge now!

fheinecke committed:
Merge branch 'master' into fred/new-yum-docs
